packages/flutter/test/cupertino/activity_indicator_test.dart

```diff
@@ -69,15 +69,79 @@ void main() {
     );
   });
 
+  testWidgets('Activity indicator 0% in progress', (WidgetTester tester) async {
+    final Key key = UniqueKey();
+    await tester.pumpWidget(
+      Center(
+        child: RepaintBoundary(
+          key: key,
+          child: Container(
+            color: CupertinoColors.white,
+            child: const CupertinoActivityIndicator.partiallyRevealed(progress: 0),
+          ),
+        ),
+      ),
+    );
+
+    await expectLater(
+      find.byKey(key),
+      matchesGoldenFile('activityIndicator.inprogress.0.0.png'),
+    );
+  });
+
+  testWidgets('Activity indicator 30% in progress', (WidgetTester tester) async {
+    final Key key = UniqueKey();
+    await tester.pumpWidget(
+      Center(
+        child: RepaintBoundary(
+          key: key,
+          child: Container(
+            color: CupertinoColors.white,
+            child: const CupertinoActivityIndicator.partiallyRevealed(progress: 0.5),
+          ),
+        ),
+      ),
+    );
+
+    await expectLater(
+      find.byKey(key),
+      matchesGoldenFile('activityIndicator.inprogress.0.3.png'),
+    );
+  });
+
+  testWidgets('Activity indicator 100% in progress', (WidgetTester tester) async {
+    final Key key = UniqueKey();
+    await tester.pumpWidget(
+      Center(
+        child: RepaintBoundary(
+          key: key,
+          child: Container(
+            color: CupertinoColors.white,
+            child: const CupertinoActivityIndicator.partiallyRevealed(progress: 1),
+          ),
+        ),
+      ),
+    );
+
+    await expectLater(
+      find.byKey(key),
+      matchesGoldenFile('activityIndicator.inprogress.1.0.png'),
+    );
+  });
+
   // Regression test for https://github.com/flutter/flutter/issues/41345.
   testWidgets('has the correct corner radius', (WidgetTester tester) async {
     await tester.pumpWidget(
       const CupertinoActivityIndicator(animating: false, radius: 100),
     );
 
+    // An early implementation for the activity indicator starting drawing
```

Comment:
```suggestion
    // An earlier implementation for the activity indicator started drawing
```

Reply:
With 
```dart
const List<int> _alphaValues = <int>[47, 47, 47, 47, 64, 81, 97, 114, 131, 147, 47, 47];
```
the starting position of the activity indicator remains the same (as the current behavior on master, the most visible segment appears at 9 o'clock initially). Since the native activity indicator also starts at 9 o'clock, I think we should keep it as is to avoid introducing breaking changes.

Reply:
I've modified the order of the values in the list so that the most prominent tick starts at 9 o'clock, and have added some comments explaining the meaning of the values. 

Regarding the "breaking" test case... there's a bit of subtlety here. This test case is designed to confirm that the rounded rectangle has been generated correctly and it happens to do so by inspecting the /first/ rrect that is painted.

My implementation draws the first rrect at 12 o'clock, whereas the old implementation used to be at 9 o'clock. However, both implementations still draw all 12 ticks... the only thing that has changed is the sequence that they get drawn in.

There are no user-facing breakages at all. This test case just happens to be very much a white-box test, and we've now changed the internal implementation details. 

Hope that makes sense.

Reply:
Ah I see. I don't think having to change this specific test makes the PR a breaking change, it was meant to ensure the corner radius scales with the activity indicator. 

What I meant by "breaking change" is the angle of the most prominent segment when `animating` is false. It used to be 9 o'clock and now its 11 o'clock. If we shift the alpha array to `[47, 47, 47, 47, 64, 81, 97, 114, 131, 147, 47, 47]` then it can go back to 9 o'clock. I doubt it will break anyone for real but it would be best if the appearance can remain exactly the same imo.

Reply:
👍 I think I shuffled those entries in an earlier commit to reflect the original behaviour. All good.